clients/gateway/gateway.go

```diff
@@ -0,0 +1,110 @@
+package gateway
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/NethermindEth/juno/utils"
+)
+
+//go:generate mockgen -destination=../../mocks/mock_gateway.go -package=mocks github.com/NethermindEth/juno/clients/gateway Gateway
+type Gateway interface {
+	AddInvokeTransaction(context.Context, *BroadcastedInvokeTxn) (*InvokeResponse, error)
+}
```

Comment:
This has been moved to the rpc package, although it remains Public so that we can use go:generate. 
If you want to keep it private and generate mocks we can still do so, but the mocks would have to be generated by hand I think.